tests/fixture/tmpnet/node_process.go

```diff
@@ -0,0 +1,258 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package tmpnet
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/ava-labs/avalanchego/api/health"
+	"github.com/ava-labs/avalanchego/config"
+	"github.com/ava-labs/avalanchego/node"
+)
+
+const (
+	AvalancheGoPathEnvName = "AVALANCHEGO_PATH"
+
+	defaultNodeInitTimeout = 10 * time.Second
+)
+
+var errNodeAlreadyRunning = errors.New("failed to start node: node is already running")
+
+func checkNodeHealth(ctx context.Context, uri string) (bool, error) {
+	// Check that the node is reporting healthy
+	health, err := health.NewClient(uri).Health(ctx, nil)
+	if err == nil {
+		return health.Healthy, nil
+	}
+
+	switch t := err.(type) {
+	case *net.OpError:
+		if t.Op == "read" {
+			// Connection refused - potentially recoverable
+			return false, nil
+		}
+	case syscall.Errno:
+		if t == syscall.ECONNREFUSED {
+			// Connection refused - potentially recoverable
+			return false, nil
+		}
+	}
+	// Assume all other errors are not recoverable
+	return false, fmt.Errorf("failed to query node health: %w", err)
+}
+
+// Defines local-specific node configuration. Supports setting default
+// and node-specific values.
+type NodeProcess struct {
+	node *Node
+
+	// PID of the node process
+	pid int
+}
+
+func (p *NodeProcess) setProcessContext(processContext node.NodeProcessContext) {
+	p.pid = processContext.PID
+	p.node.URI = processContext.URI
+	p.node.StakingAddress = processContext.StakingAddress
+}
+
+func (p *NodeProcess) readState() error {
+	path := p.getProcessContextPath()
+	if _, err := os.Stat(path); errors.Is(err, fs.ErrNotExist) {
+		// The absence of the process context file indicates the node is not running
+		p.setProcessContext(node.NodeProcessContext{})
+		return nil
+	}
+
+	bytes, err := os.ReadFile(path)
+	if err != nil {
+		return fmt.Errorf("failed to read node process context: %w", err)
+	}
+	processContext := node.NodeProcessContext{}
+	if err := json.Unmarshal(bytes, &processContext); err != nil {
+		return fmt.Errorf("failed to unmarshal node process context: %w", err)
+	}
+	p.setProcessContext(processContext)
+	return nil
+}
+
+// Start waits for the process context to be written which
+// indicates that the node will be accepting connections on
+// its staking port. The network will start faster with this
+// synchronization due to the avoidance of exponential backoff
+// if a node tries to connect to a beacon that is not ready.
+func (p *NodeProcess) Start(w io.Writer) error {
+	// Avoid attempting to start an already running node.
+	proc, err := p.getProcess()
+	if err != nil {
+		return fmt.Errorf("failed to start node process: %w", err)
```

Comment:
nit: this error message seems a bit off. It's not really starting a process here, just checking if it's running already right?